src/index.js

```diff
@@ -99,27 +117,53 @@ export function getAsnInfo(asn) {
   return { ...asns[asn], subnets: asnCidrs[asn] || [] }
 }
 
-export function getIpInfo(ip) {
+function getAsns(ip) {
   const ipPosition1 = getPosition(ip, 0)
   let ipPosition2 = getPosition(ip, 1)
   const ipInt = ipToInt(ip)
 
-  if (!rangesIndexed[ipPosition1]) {
+  if (!asnRangesIndexed[ipPosition1]) {
     return []
   }
 
-  if (!rangesIndexed[ipPosition1][ipPosition2]) {
-    const availableKeys = Object.keys(rangesIndexed[ipPosition1])
+  if (!asnRangesIndexed[ipPosition1][ipPosition2]) {
+    const availableKeys = Object.keys(asnRangesIndexed[ipPosition1])
     ipPosition2 = availableKeys.reverse().find((key) => key <= ipPosition2)
 
-    if (!rangesIndexed[ipPosition1][ipPosition2]) {
+    if (!asnRangesIndexed[ipPosition1][ipPosition2]) {
       return []
     }
   }
 
-  const filtered = rangesIndexed[ipPosition1][ipPosition2]
+  const filtered = asnRangesIndexed[ipPosition1][ipPosition2]
     .filter((range) => range && range.start <= ipInt && range.end >= ipInt)
 
   return filtered
     .map((match) => ({ ...match, ...asns[match.asn] }))
 }
+
+function getGeolocation(ip) {
+  const ipInt = ipToInt(ip)
+
+  const validRanges = []
+  for (const range of geolocationRangesIndexed) {
+    const { start, end } = range
+    if (ipInt < start) break // we can stop searching because array is sorted by start
+
+
+    if (ipInt >= start && ipInt <= end) {
+      validRanges.push(range)
```

Comment:
Surprisingly it returns multiple results. Some of them are terrible. We know accuracy, so maybe with accuracy 3, the result will be fine.
4 is accuracy on country level.